### PR TITLE
Align Context Pad

### LINKED ISSUE #114: Align Context Pad with Camunda Modeler Context Pad
__Is your feature request related to a problem? Please describe.__

Currently, the ContextPad looks different/inconsistent with the Camunda Modeler ContextPad.

**Zeebe Modeler**

**Camunda Modeler**

__Describe the solution you'd like__

It should have the same context pad items and order as the Camunda Modeler.

__Describe alternatives you've considered__

/

---

Related to https://app.prodpad.com/ideas/253/canvas.

### PR DESCRIPTION
**Pull request by oguzeroglu**
Closes #114

### CODE REVIEW

File: client/src/app/tabs/bpmn/custom/__tests__/CustomContextPadProviderSpec.js [hunk 1]
```diff
@@ -36,213 +36,228 @@ const testModules = [
...
+          'delete'
+        ]);
+      });
+    });
 
-    it('should provide SubProcess entries', inject(function() {
 
-      expectContextPadEntries('SubProcess_1', [
-        'connect',
+    it('End event', function() {
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/custom/__tests__/CustomContextPadProviderSpec.js]: as my personal feeling, I liked the 'it should ....' pattern a little more. But I would be fine with that.

File: client/src/app/tabs/bpmn/custom/__tests__/CustomContextPadProviderSpec.js [hunk 2]
```diff
@@ -36,213 +36,228 @@ const testModules = [
...
+  describe('expected entries', function() {
 
-    const diagramXML = require('./diagram.bpmn');
+    const diagramXML = require('./CustomContextPadProvider.bpmn');
 
     beforeEach(bootstrapModeler(diagramXML, { modules: testModules }));
 
-    function expectContextPadEntries(elementOrId, expectedEntries) {
+    function getContextPadEntries(elementOrId, isLabel) {
+      var entries;
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/custom/__tests__/CustomContextPadProviderSpec.js]: we can use `let` here,

File: client/src/app/tabs/bpmn/custom/CustomContextPadProvider.js [hunk 3]
```diff
@@ -45,80 +40,16 @@ export default class CustomContextPadProvider extends ContextPadProvider {
...
 
-    if (element.type === 'label') {
-      return filteredActions;
-    }
-
-    if (!is(businessObject, 'bpmn:EndEvent') && !businessObject.triggeredByEvent) {
-      if (!is(businessObject, 'bpmn:EventBasedGateway')) {
-        assign(filteredActions, { 'append.append-service-task': appendAction('bpmn:ServiceTask', 'bpmn-icon-service-task') });
-        assign(filteredActions, { 'append.append-send-task': appendAction('bpmn:ReceiveTask', 'bpmn-icon-receive-task') });
+    for (var actionKey in actions) {
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/custom/CustomContextPadProvider.js]: To be fancy, we can use es6 features here, cf. [`PaletteProvider`](https://github.com/zeebe-io/zeebe-modeler/blob/develop/client/src/app/tabs/bpmn/custom/CustomPaletteProvider.js#L46)

File: client/src/app/tabs/bpmn/custom/__tests__/CustomContextPadProviderSpec.js [hunk 4]
```diff
@@ -36,213 +36,228 @@ const testModules = [
...
 
       getBpmnJS().invoke(function(elementRegistry, contextPad) {
-
         const element = typeof elementOrId === 'string' ? elementRegistry.get(elementOrId) : elementOrId;
 
-        contextPad.open(element, true);
+        if (!isLabel) {
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/custom/__tests__/CustomContextPadProviderSpec.js]: We don't need to give this parameter into, we can determine by the element whether it's a label or not

```js
function isLabel(element) {
  return !!element.labelTarget;
}
```

For the label-text, you can then input the label element, `id = StartEvent_1_label`

File: client/src/app/tabs/bpmn/custom/modeler-options/Options.js [hunk 5]
```diff
@@ -38,11 +40,20 @@ export const AVAILABLE_REPLACE_ELEMENTS = [
 ];
 
 export const AVAILABLE_CONTEXTPAD_ENTRIES = [
+  'append.message-intermediate-event',
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/custom/modeler-options/Options.js]: good solution to overcome the previous check 👍

File: client/src/app/tabs/bpmn/custom/modeler-options/Options.js [hunk 6]
```diff
@@ -38,11 +40,20 @@ export const AVAILABLE_REPLACE_ELEMENTS = [
...
+  'append.timer-intermediate-event',
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/custom/modeler-options/Options.js]: good solution to overcome the previous check 👍

File: client/src/app/tabs/bpmn/custom/CustomContextPadProvider.js (hunk 3 above)
Comment (oguzeroglu) [reply client/src/app/tabs/bpmn/custom/CustomContextPadProvider.js]: The ordering is important here, so I implemented a fancier solution but still without the fancy spread operator :)

**Comment by pinussilvestrus** [comment]
Lets not us `feat(app): ...` but `feat(client/context-pad): ...`  for the commit message

**Pull request merged**